### PR TITLE
Closes #1224 create_single_dose_dataset

### LINKED ISSUE #1224: create_single_dose_dataset() - incorrectly deriving ASTDTM and AENDTM
### What happened?

The create_single_dose_dataset() function is correctly deriving dates for ASTDT and AENDT but not for ASTDTM and AENDTM. The function should split diary card data (where each record covers a range of dates) into one line per date. It does this correctly for ASTDT and AENDT, but it keeps the earliest recorded date for ASTDTM and AENDTM. Please see screenshot attached for example data. 

single dose
 

### Session Information

```
R version 4.0.3 (2020-10-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Red Hat Enterprise Linux

Matrix products: default
BLAS/LAPACK: /usr/lib64/libopenblas-r0.2.20.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] purrr_0.3.4       rice_2.0          lubridate_1.7.9   dplyr_1.0.2       admiraltest_0.2.0 admiral_0.7.0    

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8       rstudioapi_0.11  magrittr_2.0.3   getPass_0.2-2    hms_0.5.3        tidyselect_1.1.0 R6_2.5.0         rlang_1.0.2      stringr_1.4.0   
[10] httr_1.4.2       tools_4.0.3      cli_3.3.0        ellipsis_0.3.2   digest_0.6.27    assertthat_0.2.1 tibble_3.0.4     lifecycle_0.2.0  crayon_1.3.4    
[19] readr_1.4.0      tidyr_1.1.2      vctrs_0.4.1      curl_4.3         memoise_1.1.0    glue_1.6.2       haven_2.3.1      stringi_1.5.3    compiler_4.0.3  
[28] pillar_1.4.6     forcats_0.5.0    generics_0.1.0   jsonlite_1.7.1   pkgconfig_2.0.3
```

### Reproducible Example

```r
library(dplyr)
library(lubridate)

ex <- tibble::tribble(
  ~USUBJID, ~EXTRT, ~EXDOSFRQ, ~EXSTDTC,              ~EXENDTC,           ~ASTDT,       ~AENDT,       ~ASTDTM,               ~AENDTM,              
  "P01",    "Drug", "QD",      "2022-01-01 12:30",    "2022-01-07",       "2022-01-01", "2022-01-07", "2022-01-01 12:30:00", "2022-01-07 23:59:59",
  "P02",    "Drug", "QD",      "2022-02-06 11:09:12", "2022-02-08 12:00", "2022-02-06", "2022-02-08", "2022-02-06 11:09:12", "2022-02-08 12:00:59"
) %>% 
  mutate(
    ASTDT = ymd(ASTDT),
    AENDT = ymd(AENDT),
    ASTDTM = ymd_hms(ASTDTM),
    AENDTM = ymd_hms(AENDTM)
  )

admiral::create_single_dose_dataset(ex)
```
```
# A tibble: 10 x 9
   USUBJID EXTRT EXDOSFRQ EXSTDTC             EXENDTC          ASTDT      AENDT      ASTDTM              AENDTM             
   <chr>   <chr> <chr>    <chr>               <chr>            <date>     <date>     <dttm>              <dttm>             
 1 P01     Drug  ONCE     2022-01-01 12:30    2022-01-07       2022-01-01 2022-01-01 2022-01-01 12:30:00 2022-01-07 23:59:59
 2 P01     Drug  ONCE     2022-01-01 12:30    2022-01-07       2022-01-02 2022-01-02 2022-01-01 12:30:00 2022-01-07 23:59:59
 3 P01     Drug  ONCE     2022-01-01 12:30    2022-01-07       2022-01-03 2022-01-03 2022-01-01 12:30:00 2022-01-07 23:59:59
 4 P01     Drug  ONCE     2022-01-01 12:30    2022-01-07       2022-01-04 2022-01-04 2022-01-01 12:30:00 2022-01-07 23:59:59
 5 P01     Drug  ONCE     2022-01-01 12:30    2022-01-07       2022-01-05 2022-01-05 2022-01-01 12:30:00 2022-01-07 23:59:59
 6 P01     Drug  ONCE     2022-01-01 12:30    2022-01-07       2022-01-06 2022-01-06 2022-01-01 12:30:00 2022-01-07 23:59:59
 7 P01     Drug  ONCE     2022-01-01 12:30    2022-01-07       2022-01-07 2022-01-07 2022-01-01 12:30:00 2022-01-07 23:59:59
 8 P02     Drug  ONCE     2022-02-06 11:09:12 2022-02-08 12:00 2022-02-06 2022-02-06 2022-02-06 11:09:12 2022-02-08 12:00:59
 9 P02     Drug  ONCE     2022-02-06 11:09:12 2022-02-08 12:00 2022-02-07 2022-02-07 2022-02-06 11:09:12 2022-02-08 12:00:59
10 P02     Drug  ONCE     2022-02-06 11:09:12 2022-02-08 12:00 2022-02-08 2022-02-08 2022-02-06 11:09:12 2022-02-08 12:00:59
```

### PR DESCRIPTION
**Pull request by anthoward**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/) 
- [x] Updated relevant unit tests or have written new unit tests - See [Unit Test Guide](https://pharmaverse.github.io/admiral/articles/unit_test_guidance.html#writing-unit-tests-in-admiral-)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiral/articles/programming_strategy.html#deprecation-1)?
- [x] Update to all relevant roxygen headers and examples 
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by bms63** [comment]
@thomas-neitmann @anthoward is this something that we want to get in for this release?

**Comment by bms63** [comment]
@anthoward Are you able to fix these failing R-CMD checks?

**Comment by anthoward** [comment]
@bms63 I have made some tweaks, but not sure what's required to fix the ubuntu checks. Any ideas on what I need to do to resolve them

**Comment by bms63** [comment]
I'm not sure.  I can build it locally.  @cicdguy @thomas-neitmann Sorry to be bugging you so much today (crunch time).  Any ideas?  I can the package locally, but we are getting a can't find knirtr issue.

File: NEWS.md [hunk 1]
```diff
@@ -71,6 +71,8 @@ users feedback (#1300):
 character vector to be used for sorting the `dataset`, `derive_vars_dthcaus()` 
 updated to process additional parameter (#1125)
 
+- `create_single_dose_dataset()` Fixed bug where ASTDTM and AENDTM were not updated when start_date=ASTDT and end_date=AENDT. The function has been amended to now require start_datetime and end_datetime parameters in addition to start_date and end_date.The keep_source_vars has been added to specify the variables to be retained from the source dataset.
+
```
Comment (bms63) [inline NEWS.md:75]: Can link the issue in the news file please

**Comment by bms63** [comment]
whack a  mole today!  I don't think we had `admiral.test` available for the 4.0 and 3.6 snapshots.  I think this is why the 3.6 and 4.0 are failing here.  @cicdguy 
image

**Comment by cicdguy** [comment]
> whack a mole today! I don't think we had `admiral.test` available for the 4.0 and 3.6 snapshots. I think this is why the 3.6 and 4.0 are failing here. @cicdguy image

That makes sense. The way to get around that is to use a `staged_dependencies.yaml` file in this repo so that `admiral.test` can be installed there.

File: NEWS.md (hunk 1 above)
Comment (anthoward) [reply NEWS.md:75]: I have added the link to the issue number into the news file

**Comment by bms63** [comment]
I'm still confused on this failing check for 3.6.   @cicdguy @thomas-neitmann .  It says it can't find `admiraldev`, but the other two checks 4.0 and latest can find it.

**Comment by thomas-neitmann** [comment]
Don't let that hold us back. I've temporarily changed the 3.6 check to not be Required. So let's go ahead and merge.

**Review by bms63 — APPROVED** [review]
Go!

**Comment by bms63** [comment]
I believe the Check Templates is going to fail @thomas-neitmann

**Pull request merged**